src/jpeg/encoder.rs

```diff
@@ -736,47 +742,39 @@ fn build_frame_header(
 fn build_scan_header(m: &mut Vec<u8>, components: &[Component]) {
     m.clear();
 
-    // TODO: More idiomatic would be extend_from_slice, to_be_bytes
-    let _ = m.write_all(&[components.len() as u8]);
+    let _ = m.push(components.len() as u8);
```

Comment:
`let _ = ` should no longer be necessary. I like this change.

Reply:
I realize I might have been unclear here. Please remove the `let _ = ` part, keep the test.

Reply:
I missed that, but there should be no more `let _ = ` in encoder.rs.